### PR TITLE
feat: add groupLabel prop in fieldNames

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix https://github.com/ant-design/ant-design/issues/42328
- fix https://github.com/react-component/select/pull/942

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     add groupLabel prop in fieldNames      |
| 🇨🇳 Chinese |      fieldNames 新增 groupLabel 字段功能     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 82e9732</samp>

Added support for customizing the group label field name in the `Select` and `OptGroup` components. Updated the documentation and the `rc-select` dependency accordingly.

### <samp>🤖 Generated by Copilot at 82e9732</samp>

*  Added a new `fieldNames.groupLabel` prop to the `Select` component to allow customizing the group label field name in the `OptGroup` component. ([link](https://github.com/ant-design/ant-design/pull/42492/files?diff=unified&w=0#diff-9c8a1d9a817ce1e6f60bd9a83aeb6cc9ef87068d93611aa38f6dd25a2093827bL3-R7), [link](https://github.com/ant-design/ant-design/pull/42492/files?diff=unified&w=0#diff-bf56ff1410a74a10dffb8c98ad2163749383eb3d25cc89fd10e747bbd8004452L73-R73), [link](https://github.com/ant-design/ant-design/pull/42492/files?diff=unified&w=0#diff-07bf0b280d743be2023d98c096d6127d8fa2d5e9baa0c23ab8ee2a4f1c0db96bL74-R74))
* Updated the `rc-select` dependency version to `~14.5.0` in the `package.json` file to use the latest version of the `rc-select` component that supports the `fieldNames.groupLabel` prop. ([link](https://github.com/ant-design/ant-design/pull/42492/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L144-R144))

### CODE REVIEW

File: components/select/index.en-US.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ Select component to select value from options.
 | popupMatchSelectWidth | Determine whether the popup menu and the select input are the same width. Default set `min-width` same as input. Will ignore when value less than select width. `false` will disable virtual scroll | boolean \| number | true | 5.5.0 |
 | dropdownRender | Customize dropdown content | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | The style of dropdown menu | CSSProperties | - |  |
-| fieldNames | Customize node label, value, options field name | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | Customize node groupLabel label, value, options field name | object | { groupLabel: `lanel`, label: `label`, value: `value`, options: `options` } | 4.17.0 |
```
Comment (Wxh16144) [inline components/select/index.en-US.md:73]: `lanel` => `label` ?

Comment (BoyYangzai) [reply components/select/index.en-US.md:73]: @Wxh16144 呜呜呜是的

**Comment by BoyYangzai** [comment]
这个版本还有点问题 先等fix

File: components/select/index.zh-CN.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ demo:
 | popupMatchSelectWidth | 下拉菜单和选择器同宽。默认将设置 `min-width`，当值小于选择框宽度时会被忽略。false 时会关闭虚拟滚动 | boolean \| number | true | 5.5.0 |
 | dropdownRender | 自定义下拉框内容 | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | 下拉菜单的 style 属性 | CSSProperties | - |  |
-| fieldNames | 自定义节点 label、value、options 的字段 | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | 自定义节点 groupLabel label、value、options 的字段 | object | { groupLabel: `label`, label: `label`, value: `value`, options: `options` } | 4.17.0 |
 | filterOption | 是否根据输入项进行筛选。当其为一个函数时，会接收 `inputValue` `option` 两个参数，当 `option` 符合筛选条件时，应返回 true，反之则返回 false | boolean \| function(inputValue, option) | true |  |
```
Comment (Wxh16144) [inline components/select/index.zh-CN.md:75]: 现在默认值 groupLabel 是 label 吗? 还是没有 groupLabel 默认值? 这个感觉没有说明 groupLabel 是新增的, 后面的 4.17.0 可能会造成歧义

File: components/select/demo/optgroup.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## zh-CN
 
-用 `OptGroup` 进行选项分组。
+用 `OptGroup` 进行选项分组。分组 `label` 字段名修改优先级: `fieldNames.groupLbael` > `fieldNames.label` > `label`
```
Comment (Wxh16144) [inline components/select/demo/optgroup.md:3]: 个人观点是不需要再这里加描述, 直接 API 里面说明就好了, 不过还是需要等核心成员 review 一遍

File: components/select/index.zh-CN.md (hunk 2 above)
Comment (BoyYangzai) [reply components/select/index.zh-CN.md:75]: @Wxh16144 是新增的 4.17.0是没有的 我也在想怎么处理

File: components/select/demo/optgroup.md (hunk 3 above)
Comment (BoyYangzai) [reply components/select/demo/optgroup.md:3]: 好～

File: components/select/index.en-US.md [hunk 4]
```diff
@@ -70,7 +70,7 @@ Select component to select value from options.
 | popupMatchSelectWidth | Determine whether the popup menu and the select input are the same width. Default set `min-width` same as input. Will ignore when value less than select width. `false` will disable virtual scroll | boolean \| number | true | 5.5.0 |
 | dropdownRender | Customize dropdown content | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | The style of dropdown menu | CSSProperties | - |  |
-| fieldNames | Customize node label, value, options field name | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | Customize node groupLabel label, value, options field name | object | { groupLabel: `label`, label: `label`, value: `value`, options: `options` } | 4.17.0 |
```
Comment (afc163) [inline components/select/index.en-US.md:73]:
```suggestion
| fieldNames | Customize node `label`, `value` and `groupLabel` options field name | object | { groupLabel: `label`, label: `label`, value: `value`, options: `options` } | 4.17.0 |
```

File: components/select/index.zh-CN.md [hunk 5]
```diff
@@ -71,7 +71,7 @@ demo:
 | popupMatchSelectWidth | 下拉菜单和选择器同宽。默认将设置 `min-width`，当值小于选择框宽度时会被忽略。false 时会关闭虚拟滚动 | boolean \| number | true | 5.5.0 |
 | dropdownRender | 自定义下拉框内容 | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | 下拉菜单的 style 属性 | CSSProperties | - |  |
-| fieldNames | 自定义节点 label、value、options 的字段 | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | 自定义节点 groupLabel label、value、options 的字段 | object | { groupLabel: `label`, label: `label`, value: `value`, options: `options` } | 4.17.0 |
```
[the remark this answers is not in the document]
Comment (BoyYangzai) [reply components/select/index.zh-CN.md:74]: 👌

File: components/select/demo/optgroup.md (hunk 3 above)
Comment (BoyYangzai) [reply components/select/demo/optgroup.md:3]: @afc163 偏右大佬 这里怎么说

File: components/select/index.en-US.md (hunk 1 above)
Comment (RedJue) [reply components/select/index.en-US.md]: 建议装个拼写的插件🤔

File: components/select/index.en-US.md [hunk 6]
```diff
@@ -70,7 +70,7 @@ Select component to select value from options.
 | popupMatchSelectWidth | Determine whether the popup menu and the select input are the same width. Default set `min-width` same as input. Will ignore when value less than select width. `false` will disable virtual scroll | boolean \| number | true | 5.5.0 |
 | dropdownRender | Customize dropdown content | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | The style of dropdown menu | CSSProperties | - |  |
-| fieldNames | Customize node label, value, options field name | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | Customize node label, value, options，groupLabel field name | object | { label: `label`, value: `value`, options: `options`, groupLabel: `label` } | 4.17.0（ groupLabel added in 5.6.0 ） |
```
Comment (MadCcc) [inline components/select/index.en-US.md:73]:
```suggestion
| fieldNames | Customize node label, value, options，groupLabel field name | object | { label: `label`, value: `value`, options: `options`, groupLabel: `label` } | 4.17.0 (groupLabel added in 5.6.0) |
```

File: components/select/index.zh-CN.md [hunk 7]
```diff
@@ -71,7 +71,7 @@ demo:
 | popupMatchSelectWidth | 下拉菜单和选择器同宽。默认将设置 `min-width`，当值小于选择框宽度时会被忽略。false 时会关闭虚拟滚动 | boolean \| number | true | 5.5.0 |
 | dropdownRender | 自定义下拉框内容 | (originNode: ReactNode) => ReactNode | - |  |
 | dropdownStyle | 下拉菜单的 style 属性 | CSSProperties | - |  |
-| fieldNames | 自定义节点 label、value、options 的字段 | object | { label: `label`, value: `value`, options: `options` } | 4.17.0 |
+| fieldNames | 自定义节点 label、value、options、groupLabel 的字段 | object | { label: `label`, value: `value`, options: `options`, groupLabel: `label` } | 4.17.0（groupLabel 在 5.6.0 新增） |
```
Comment (MadCcc) [inline components/select/index.zh-CN.md:74]:
```suggestion
| fieldNames | 自定义节点 label、value、options、groupLabel 的字段 | object | { label: `label`, value: `value`, options: `options`, groupLabel: `label` } | 4.17.0（`groupLabel` 在 5.6.0 新增） |
```

File: components/select/demo/optgroup.md (hunk 3 above)
Comment (MadCcc) [reply components/select/demo/optgroup.md:3]: 确实不用写在这里，可以在 API 处就一个 `fieldNames` 的表格

Comment (zombieJ) [reply components/select/demo/optgroup.md:3]: 不需要特别标注 +1

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**